forui/lib/src/widgets/badge/badge.dart

```diff
@@ -0,0 +1,172 @@
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+
+import 'package:meta/meta.dart';
+
+import 'package:forui/forui.dart';
+
+part 'badge_content.dart';
+part 'badge_styles.dart';
+
+/// A badge, or a component that looks like a badge.
+class FBadge extends StatelessWidget {
+
+  /// The design.
+  final FBadgeDesign design;
+  /// A callback for when the badge is pressed.
+  final void Function(BuildContext)? onPressed;
+  /// A callback for when the badge is long pressed.
+  final void Function(BuildContext)? onLongPressed;
+  /// The builder.
+  final Widget Function(BuildContext, FBadgeStyle) builder;
+
+  /// Creates a [FBadge].
+  FBadge(String label, {required this.design, this.onPressed, this.onLongPressed, super.key}):
```

Comment:
Should we make `label` named? I think it's more within flutter's general convention.